src/body.rs

```diff
@@ -62,6 +63,47 @@ where
     }
 }
 
+#[cfg(feature = "hyper-body")]
+#[allow(unconditional_recursion)]
+impl<T> ::hyper::body::Payload for BoxBody<T>
+where
+    T: bytes::Buf + Send + IntoBuf + 'static,
+{
+    type Data = T;
+    type Error = ::hyper::Error;
+
+    fn poll_data(&mut self) -> Poll<Option<Self::Data>, Self::Error> {
+        ::hyper::body::Payload::poll_data(self)
```

Comment:
this should call `Body::poll_data(self)` I think, this is a recusive call because you are implementing payload and then calling yourself. You want to call the `tower_grpc::body::Body` trait impl i think. Thats how tower-h2 does it.